03-Purpose-and-Principles/Documentation-Principles.md

```diff
@@ -0,0 +1,46 @@
+# Documentation Principles
+
+These principles are not set in stone -- if you'd like clarification, qualification, addition, removal, or changes to these principles, open an issue to discuss.
+
+## Documentation Scope
+
+- There are many types of applications that can be built with PureScript, each of which should have documentation. The group of people who write each type of application is best suited for managing its documentation.
+- All types of PureScript applications have pre-requisite knowledge, and the set of this knowledge which is common across all PureScript applications should be managed by the group of PureScript groups, an entity which we will name the PureScript organization.
+- The current scope of this project is the documentation managed by the PureScript organization. Other efforts can be responsible for documentation managed by PureScript groups.
+
+## Documentation Structure
+
+- Documentation is easiest to maintain if it follows a well-defined format. Templates offer a simple way of starting a new document.
+- Documentation is easiest to maintain if it clearly defines its audience, its goal, and the primary terms it uses.
+
+## Documentation Content
+
+- *Some* documentation is better than no documentation.
+- Documentation is easiest for a learner to navigate when its structure uses terms in the problem domain. Its content, then, is free to use terms in the solution domain.
+- Documentation should avoid implicit bias towards a specific project as a solution. Instead, documentation should clearly explain the problem and only refer to specific projects in a list of implementations of the solution. While it's important to have a default recommended solution in a community, it's also important to give people freedom to add their own novel solutions to the ecosystem.
```

Comment:
I think this depends on the type of documentation. If someone was writing a 'how-to' guide for a library, one likely own't be referring to other implementations in that documentation.

Reply:
That's true, I wasn't thinking about libraries... I suppose I would want to redesign this principle to suit libraries like "Content Should Teach, Not Prescribe". For libraries, this would advise doc writers to *explain* why the functions are designed as they are rather than just *showing* how to use the functions. What do you think - would this principle phrased like this satisfy that situation?

Reply:
I revised that principle to clarify my intent. Here is the new version, for convenience.

```
- **Content Should Teach, Not Prescribe**: Documentation should guide the audience, not simply give an answser or prescribe a solution. While it's important to have a default solution in a community to recommend, it's also important to ensure people feel free to contribute new solutions to the ecosystem.
```

Is your concern here alleviated?

Reply:
Your raising this point up was great - it was pretty poorly expressed before.

Reply:
> Is your concern here alleviated?

I believe so.